src/test/serializable/base.py

```diff
@@ -0,0 +1,46 @@
+from copy import deepcopy
+
+
+class SerializableTestMixin(object):
+    """
+    Mixin class for testing sub-classes of Serializable.
+    """
+
+    def test_roundtrip(self) -> None:
+        _dict = self._object.to_dict()
+        _recreated_object = self._object.__class__.from_dict(_dict)
+        self.assertTrue(
+            self._object == _recreated_object,
+            msg="Roundtrip failed for %s: recreated object not equal." % self._object.__class__.__name__
+        )
+
+    def test_from_dict(self) -> None:
+        _read_object = self._object.__class__.from_dict(self._object_dict)
+        self.assertTrue(
+            self._object == _read_object,
+            msg="Reading in object failed for %s: read in object not equal." % self._object.__class__.__name__
+        )
+
+    def test_from_other_dict(self) -> None:
+        _read_object = self._object.__class__.from_dict(self._object_dict)
+        _other_read_object = self._object.__class__.from_dict(self._other_object_dict)
+        self.assertFalse(
+            _read_object == _other_read_object,
+            msg="Reading in different objects failed for %s: read in objects are the same."
+                % self._object.__class__.__name__
+        )
+
+    def test_removing_property_raises_error(self) -> None:
+        for _key in list(self._object_dict.keys()):
+            _object_dict_copy = deepcopy(self._object_dict)
+            _object_dict_copy.pop(_key)
+            if _key in self._optional_properties:
+                self._object.__class__.from_dict(_object_dict_copy)
+            else:
+                with self.assertRaises(ValueError):
+                    self._object.__class__.from_dict(_object_dict_copy)
+
+    def test_adding_property_raises_error(self) -> None:
+        self._object_dict["AYY"] = "LAMO"
```

Comment:
`LMAO`  lol

Reply:
lol

Reply:
LAMO is the ascended version of the meme.

Reply:
would like to see a reference to this ascended version :P

Reply:
or maybe we make it ascend, forget about it

Reply:
Can't give a reference because the usage wasn't logged.
The joke is that you would intentionally spell it "ayy lamo" to spot anyone new.